src/coreComponents/fileIO/vtk/VTKVTMWriter.cpp

```diff
@@ -59,11 +59,18 @@ void VTKVTMWriter::addSubBlock( string const & blockName, string const & subBloc
   subBlockNode.append_attribute( "name" ) = subBlockName.c_str();
 }
 
-void VTKVTMWriter::addDataToSubBlock( string const & blockName, string const & subBlockName, string const & filePath, int mpiRank ) const
+void VTKVTMWriter::addSubSubBlock( string const & blockName, string const & subBlockName, string const & subSubBlockName ) const
 {
-  auto blockNode = m_vtmFile.child( "VTKFile" ).child( "vtkMultiBlockDataSet" ).find_child_by_attribute( "Block", "name", blockName.c_str() );
-  auto subBlockNode = blockNode.find_child_by_attribute( "Block", "name", subBlockName.c_str() );
-  auto dataNode = subBlockNode.append_child( "DataSet" );
+  auto subBlockNode = m_vtmFile.child( "VTKFile" ).child( "vtkMultiBlockDataSet" ).find_child_by_attribute( "Block", "name", blockName.c_str() ).find_child_by_attribute( "Block", "name", subBlockName.c_str() );
+  auto subSubBlockNode = subBlockNode.append_child( "Block" );
+  subSubBlockNode.append_attribute( "name" ) = (blockName + "_" + subSubBlockName).c_str();
```

Comment:
The same value `blockName + "_" + subSubBlockName` is used in both `addSubSubBlock` and `addDataToSubSubBlock`.
If it's supposed to be the same, could you build a unique function that builds the name from `blockName` and `subSubBlockName`.

Reply:
I'm not sure I understand - are you suggesting to merge those two functions into a single function?

Reply:
The core issue is that your code relies on the naming pattern of your sub-block to be `blockName + "_" + subSubBlockName`.
So this pattern generation must be unique (if you change it, you change it everywhere, and the code still works).

In your current implementation, you can change the first `_` to `-`, forgetting to modify the second by design. It will compile but the code won't work anymore.

* A solution is to centralize this naming pattern generation in its own function.
* Another solution could be (i'm not 100% sure) to make one unique function `getOrCreateSubSubBlock` that could help the factorisation. I think that the naming pattern generation would be used only once in this function and so you would not need an extra function just for this. Up to you 😉